optimizer_wrapper.rb

```diff
@@ -221,14 +221,27 @@ def self.solve(service_vrp, job = nil, block = nil)
         end
       }
 
+      # vrp.periodic_heuristic check the first_solution_stategy which may change right after periodic heuristic
+      periodic_heuristic_flag = vrp.periodic_heuristic?
       # TODO: refactor with dedicated class
       if vrp.schedule?
         periodic = Interpreters::PeriodicVisits.new(vrp)
         vrp = periodic.expand(vrp, job, &block)
         optim_solution = vrp.configuration.preprocessing.heuristic_result if vrp.periodic_heuristic?
+        if vrp.configuration.resolution.solver && vrp.configuration.preprocessing.first_solution_strategy == ['periodic']
+          vrp.configuration.resolution.duration -= optim_solution.elapsed
+          if vrp.configuration.resolution.minimum_duration && vrp.configuration.resolution.duration
+            vrp.configuration.resolution.minimum_duration *= 1 - optim_solution.elapsed / vrp.configuration.resolution.duration
+          end
+
+          if vrp.configuration.resolution.solver && vrp.periodic_heuristic?
+            vrp.configuration.preprocessing.first_solution_strategy  -= ['periodic']
+            vrp.configuration.preprocessing.first_solution_strategy  += ['global_cheapest_arc'] if vrp.configuration.preprocessing.first_solution_strategy.empty?
+          end
+        end
```

Comment:
I don't think we should modify the time. The time passed in periodic has nothing to do with solver time -- because it is not controlled by minimum_and maximum_duration and as we do not modify the minimum/maximum duration due to other preprocessing as Split/Dicho time passed inside vroom/clustering, we should leave this time be.

The intermediary optimization time is not controlled by the user and can be excluded from the calculations. Otherwise, the user has no control over how much time will be spent inside the last optimisation. 

It looks like my previous suggestion to have an intermediary variable failed due to the assignment operator not modifying the original array :face_exhaling:  sorry about that. What about the one down below, it is a lot more compact and doesn't generate a future Rubocop violation :heart: 
```suggestion
        if vrp.periodic_heuristic?
          optim_solution = vrp.configuration.preprocessing.heuristic_result

          if vrp.configuration.resolution.solver
            first_solution_strategy = vrp.configuration.preprocessing.first_solution_strategy
            first_solution_strategy.delete('periodic')
            first_solution_strategy << 'global_cheapest_arc' if first_solution_strategy.empty?
          end
        end
```

Reply:
Oh my bad ! I was pretty sure i've removed the time reduction